app/src/main/java/ch/sdp/vibester/auth/FireBaseAuthenticator.kt

```diff
@@ -25,17 +29,58 @@ class FireBaseAuthenticator @Inject constructor() {
          * API: return the mail of the user if logged in otherwise empty string
          */
         fun getCurrentUserMail(): String {
+            var mail = ""
             if (isLoggedIn()) {
-                return FirebaseAuth.getInstance().currentUser!!.email.toString()
+                mail = FirebaseAuth.getInstance().currentUser!!.email.toString()
+            }
+            return mail
+        }
+
+        fun getCurrentUID(): String {
+            var uid = ""
+            if (isLoggedIn()) {
+                uid = FirebaseAuth.getInstance().currentUser!!.uid
+            }
+            return uid
+        }
+
+
+        /**
+         * A function to return the result of google sign in
+         * @param requestCode a request code
+         * @param resultCode a result code
+         * @param data intent returned from google sign in
+         */
+        fun googleActivityResult(requestCode: Int, resultCode: Int, data: Intent?, ctx: Context): String? {
+            if (requestCode == 1000) {
```

Comment:
request code 1000 appears more than once in different files, it might be useful to store it as a const and give it an appropriate name.